hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeResourcePolicy.java

```diff
@@ -73,6 +78,11 @@ static boolean areResourcesAvailable(
       // required resources available.
       return requiredResourceCount > 0;
     } else {
+      if (minimumRedundantResources > resources.size()){
+        LOG.info("Resource not available. Details: redundantResourceCount=" + redundantResourceCount
+            + ", disabledRedundantResourceCount=" + disabledRedundantResourceCount
+            + ", minimumRedundantResources=" + minimumRedundantResources + ".");
+      }
       return redundantResourceCount - disabledRedundantResourceCount >=
           minimumRedundantResources;
```

Comment:
Change it to 
```

      final boolean areResourceAvailable =
          redundantResourceCount - disabledRedundantResourceCount >= minimumRedundantResources;
      if (!areResourceAvailable) {
        LOG.info("Resources not available. Details: redundantResourceCount={},"
                + " disabledRedundantResourceCount={}, minimumRedundantResources={}.",
            redundantResourceCount, disabledRedundantResourceCount, minimumRedundantResources);
      }
      return areResourceAvailable;
```

Reply:
I've changed the code and it's passing the tests locally, thanks for your help!!!